Checkout/Checkout/Test/Validation/CardNumberValidatorTests.swift

```diff
@@ -83,7 +83,85 @@ final class CardNumberValidatorTests: XCTestCase {
     ]
 
     testCases.forEach { cardNumber, expectedResult in
-      let actualResult = subject.validate(cardNumber: cardNumber)
+      let actualResult = subject.eagerValidate(cardNumber: cardNumber)
+      XCTAssertEqual(
+        actualResult,
+        expectedResult,
+        "expected \(expectedResult) for card number \(cardNumber), received \(actualResult)"
+      )
+
+      XCTAssertEqual(stubLuhnChecker.luhnCheckCalledWith, nil)
+    }
+  }
+
+  func test_eagerValidateCardNumber_multipleValidLengths() {
```

Comment:
Can we have invalid values which are copy pasted garbage entries like below as one more test ?  

 "-42": .success(.visa),
      "$%424": .success(.visa),
      "werewr4242": .success(.visa),
      "42424-": .success(.visa),
      "424242asdf": .success(.visa),